lib/api/environment.rb

```diff
@@ -1,7 +1,30 @@
 module Api
   class Environment
     def self.normalized_attributes
-      @normalized_attributes ||= {:time => {}, :url => {}, :resource => {}, :encrypted => {}}
+      @normalized_attributes ||= {
+        :time      => time_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = true },
```

Comment:
Out of curiosity, is the `true` value used at all?  If not, perhaps it would be better to use a Set over a Hash in all of these cases.